router/usecase/pools/routable_cw_alloy_transmuter_pool_test.go

```diff
@@ -0,0 +1,253 @@
+package pools_test
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/osmosis-labs/sqs/domain"
+	"github.com/osmosis-labs/sqs/domain/mocks"
+	"github.com/osmosis-labs/sqs/router/usecase/pools"
+	"github.com/osmosis-labs/sqs/sqsdomain"
+
+	"github.com/osmosis-labs/osmosis/osmomath"
+)
+
+func (s *RoutablePoolTestSuite) SetupRoutableAlloyTransmuterPool(tokenInDenom, tokenOutDenom string, balances sdk.Coins) sqsdomain.RoutablePool {
+	cosmwasmPool := s.PrepareCustomTransmuterPool(s.TestAccs[0], []string{tokenInDenom, tokenOutDenom})
+
+	poolType := cosmwasmPool.GetType()
+
+	veryBigNormalizationFactor, _ := osmomath.NewIntFromString("999999999999999999999999999999999999999999")
+
+	mock := &mocks.MockRoutablePool{
+		ChainPoolModel: cosmwasmPool.AsSerializablePool(),
+		CosmWasmPoolModel: sqsdomain.NewCWPoolModel(
+			"crates.io:transmuter", "3.0.0",
+			sqsdomain.CWPoolData{
+				AlloyTransmuter: &sqsdomain.AlloyTransmuterData{
+					AlloyedDenom: ALLUSD,
+					AssetConfigs: []sqsdomain.TransmuterAssetConfig{
+						{Denom: USDC, NormalizationFactor: osmomath.NewInt(100)},
+						{Denom: USDT, NormalizationFactor: osmomath.NewInt(1)},
+						{Denom: "overlypreciseusd", NormalizationFactor: veryBigNormalizationFactor},
+						{Denom: "noprecisionusd", NormalizationFactor: osmomath.ZeroInt()},
+						{Denom: ALLUSD, NormalizationFactor: osmomath.NewInt(10)},
+					},
+				},
+			},
+		),
+		Balances: balances,
+		PoolType: poolType,
+	}
+
+	routablePool, err := pools.NewRoutablePool(mock, tokenOutDenom, noTakerFee, domain.CosmWasmPoolRouterConfig{
+		IsAlloyedTransmuterEnabled: true,
+	})
+	s.Require().NoError(err)
+
+	return routablePool
+}
+
+// Tests no slippage quotes and validation edge cases aroun transmuter pools.
+func (s *RoutablePoolTestSuite) TestCalculateTokenOutByTokenIn_AlloyTransmuter() {
+	defaultBalances := sdk.NewCoins(sdk.NewCoin(USDC, osmomath.NewInt(1000000)), sdk.NewCoin(USDT, osmomath.NewInt(1000000)))
+
+	tests := map[string]struct {
+		tokenIn     sdk.Coin
+		tokenOut    sdk.Coin
+		balances    sdk.Coins
+		expectError error
+	}{
+		"valid transmuter quote": {
+			tokenIn:  sdk.NewCoin(USDT, osmomath.NewInt(10000)),
+			tokenOut: sdk.NewCoin(USDC, osmomath.NewInt(1000000)),
```

Comment:
nit: ditto for vars for int values to simplify future updates and reference checking.

Applies to other tests as well.